x/amm/keeper/update_pool_for_swap.go

```diff
@@ -75,21 +75,31 @@ func (k Keeper) UpdatePoolForSwap(
 
 	swapFeeCoins := swapFeeInCoins.Add(swapFeeOutCoins...)
 
-	// calculate treasury amount to send as bonus
-	rebalanceTreasuryAddr := sdk.MustAccAddressFromBech32(pool.GetRebalanceTreasury())
-	treasuryTokenAmount := k.bankKeeper.GetBalance(ctx, rebalanceTreasuryAddr, tokenOut.Denom).Amount
-	bonusTokenAmount := sdk.NewDecFromInt(tokenOut.Amount).Mul(weightBalanceBonus).RoundInt()
-	if treasuryTokenAmount.LT(bonusTokenAmount) {
-		bonusTokenAmount = treasuryTokenAmount
-	}
+	if weightBalanceBonus.IsPositive() {
+		rebalanceTreasuryAddr := sdk.MustAccAddressFromBech32(pool.GetRebalanceTreasury())
+		bonusTokenAmount := sdk.NewDecFromInt(tokenOut.Amount).Mul(weightBalanceBonus).RoundInt()
 
-	// send bonus tokens to recipient if positive
-	if weightBalanceBonus.IsPositive() && bonusTokenAmount.IsPositive() {
+		// send bonusTokenAmount from pool addr to treasury addr
 		bonusToken := sdk.NewCoin(tokenOut.Denom, bonusTokenAmount)
-		err = k.bankKeeper.SendCoins(ctx, rebalanceTreasuryAddr, recipient, sdk.Coins{bonusToken})
+		err = k.bankKeeper.SendCoins(ctx, poolAddr, rebalanceTreasuryAddr, sdk.Coins{bonusToken})
 		if err != nil {
 			return sdk.ZeroInt(), err
 		}
+
+		// calculate treasury amount to send as bonus
+		treasuryTokenAmount := k.bankKeeper.GetBalance(ctx, rebalanceTreasuryAddr, tokenOut.Denom).Amount
+		if treasuryTokenAmount.LT(bonusTokenAmount) {
+			bonusTokenAmount = treasuryTokenAmount
```

Comment:
This validation should be unnecessary right? As we are sending the amount above, it's guaranteed it will be true always?

Reply:
correct, we could also very well send the bonus token amount from the pool addr directly to the recipient without relying on the treasury address right?

Reply:
Yes, but let's just keep it this way. Makes easy to track

Reply:
I added a comment to clarify the direct transfer from pool addr to treasury to avoid any confusions